src/app/shared/pipes/authors.pipe.ts

```diff
@@ -0,0 +1,29 @@
+import { Pipe, PipeTransform } from '@angular/core';
+import { CoursesStoreService } from '@app/services/courses-store.service';
+
+import { Author } from '@shared/models/author.model';
+
+@Pipe({
+  name: 'authors',
+  pure: false,
```

Comment:
Although you can make a pipe impure, that is not at all desired, especially not in our case. Can you please refactor by removing the dependency with the CoursesStore?
Hint: you can add multiple pipes on your templates if needed, and start with the async pipe for observables

Reply:
Nice tip, i was trying to figure out how should i do it but nothing worked out. Thanks